### PR TITLE
Eng 1123 enable operator with no return

### PR DESCRIPTION
**Pull request by Boyuan-Deng**
## Describe your changes and why you are making these changes

Before this implementation, is the user declared a @op for a function with null return, our system will produce a error. I wrote the code to enable operator with no return type. 

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: sdk/aqueduct/artifacts/none_artifact.py
Comment (kenxu95) [inline sdk/aqueduct/artifacts/none_artifact.py:15]: This entire comment is irrelevant to NoneArtifact right.

File: sdk/aqueduct/artifacts/none_artifact.py
Comment (kenxu95) [inline sdk/aqueduct/artifacts/none_artifact.py:39]: this method doesn't raise anything...

Comment (kenxu95) [reply sdk/aqueduct/artifacts/none_artifact.py:39]: Just say it returns None.

File: sdk/aqueduct/artifacts/none_artifact.py
Comment (kenxu95) [reply sdk/aqueduct/artifacts/none_artifact.py:15]: I'm curious if we need a separate NoneArtifact, or can we do everything in GenericArtifact? @cw75 I know you use content == None, but I wonder if we could just allow for content to be Optional and set other internal fields. Conceptually, this artifact seems unnecessary to me.

File: sdk/aqueduct/deserialize.py
Comment (kenxu95) [inline sdk/aqueduct/deserialize.py:38]: shouldn't this return None?

Comment (kenxu95) [reply sdk/aqueduct/deserialize.py:38]: To check my understanding, `content` is always an empty bytestring here right?

File: src/python/aqueduct_executor/operators/utils/utils.py
Comment (kenxu95) [inline src/python/aqueduct_executor/operators/utils/utils.py:181]: grammar/style nit: Capitalize "function", uncapitalize "Wrong".

File: src/python/aqueduct_executor/operators/utils/utils.py
Comment (kenxu95) [inline src/python/aqueduct_executor/operators/utils/utils.py:305]: Better to just do `elif value is None`

File: src/ui/common/src/components/workflows/nodes/NoneArtifactNode.tsx
Comment (kenxu95) [inline src/ui/common/src/components/workflows/nodes/NoneArtifactNode.tsx:13]: nit: remove this before merging. 

Do you have a screenshot of this?

**Review by kenxu95 — CHANGES_REQUESTED** [review]
Nice! Some comments and one discussion topic.

File: sdk/aqueduct/deserialize.py
Comment (Boyuan-Deng) [reply sdk/aqueduct/deserialize.py:38]: yes, _read_none_content should always be a empty bytes(). And you are right on that _read_none_artifact should always return None

File: src/ui/common/src/components/workflows/nodes/NoneArtifactNode.tsx
Comment (Boyuan-Deng) [reply src/ui/common/src/components/workflows/nodes/NoneArtifactNode.tsx:13]: image

File: sdk/aqueduct/artifacts/none_artifact.py
Comment (Boyuan-Deng) [reply sdk/aqueduct/artifacts/none_artifact.py:15]: I discussed this with Cg on last friday and we decided to proceed with creating a separate NoneArtifact mainly because GenericArtifact is a temporary solution for artifact types that we did not get to come up with meaning type-specific functions eg( head() for table artifact). I think in the future, Cg wants to slowly pull artifact type out of the GenericAritifact type and assign them type specific functions. From my personal standpoint, NoneArtifact and GenericArtifact shows the same code implementation so it was kinda redundant.

Comment (cw75) [reply sdk/aqueduct/artifacts/none_artifact.py]: As long as we can distinguish cases where the content is expected to be None v.s. the content is None because we haven't evaluated the content yet (constructing the artifact class from a flow run), I'm ok with merging.

File: src/python/aqueduct_executor/operators/utils/utils.py
Comment (kenxu95) [inline src/python/aqueduct_executor/operators/utils/utils.py:178]: The assumption here is that output is None already for our inference, so we shouldn't even check it. Can you just remove `output` completely and just have this be a single `storage.put()` line like the others?

**Review by kenxu95 — DISMISSED** [review]
Where are the integration tests for this?

File: sdk/aqueduct/artifacts/param_artifact.py
Comment (Boyuan-Deng) [reply sdk/aqueduct/artifacts/param_artifact.py:21]: No, this does not relate to this PR but notice it for being inconsistent with other artifact. Will remove it from this PR and ping chenggang for this change

File: sdk/aqueduct/artifacts/none_artifact.py
Comment (kenxu95) [reply sdk/aqueduct/artifacts/none_artifact.py]: I think we can, since we can use the `from_flow_run` flag to check the second case.

**Comment by Boyuan-Deng** [comment]
> Where are the integration tests for this?

Done

File: src/ui/common/src/components/workflows/nodes/NoneArtifactNode.tsx
Comment (agiron123) [inline src/ui/common/src/components/workflows/nodes/NoneArtifactNode.tsx:15]: @vsreekanti is this the icon that we wish to use?

Perhaps we can use some sort of "function" or "computation" icon here?

**Review by agiron123 — APPROVED** [review]
FE parts here look good to me. Nice work!

**Comment by cw75** [comment]
I'll merge this after making sure it works well with the lazy execution stuff.

**Pull request closed without merging**
